File: ppt/my-ai-ppt/apps/api/app/schemas/dsl.py
from typing import Any, Dict, List, Optional
from pydantic import BaseModel, Field


class SlideContent(BaseModel):
    title: Optional[str] = None
    subtitle: Optional[str] = None
    body: Optional[str] = None
    footer: Optional[str] = None
    bullets: List[str] = Field(default_factory=list)
    image_src: Optional[str] = None


class Slide(BaseModel):
    slide_id: str
    page_type: str
    layout_id: str
    theme_override: Optional[Dict[str, Any]] = None
    content: SlideContent
    constraints: Dict[str, Any] = Field(default_factory=dict)


class ProjectDSL(BaseModel):
    project_id: str
    theme: Dict[str, Any] = Field(default_factory=dict)
    slides: List[Slide] = Field(default_factory=list)

File: ppt/my-ai-ppt/packages/shared-types/dsl.ts
export type Slide = {
  slide_id: string;
  page_type: string;
  layout_id: string;
  theme_override?: Record<string, unknown> | null;
  content: {
    title?: string;
    subtitle?: string;
    body?: string;
    footer?: string;
    bullets?: string[];
    image_src?: string | null;
  };
  constraints?: Record<string, unknown>;
};

export type ProjectDSL = {
  project_id: string;
  theme: Record<string, unknown>;
  slides: Slide[];
};
